src/chat/ChatScreen.js

```diff
@@ -188,6 +188,7 @@ export default function ChatScreen(props: Props) {
           onSend={sendCallback}
           defaultTopic={editMessage ? editMessage.topic : undefined}
           defaultMessage={editMessage ? editMessage.content : draft}
+          autoFocusMessage={editMessage !== null}
```

Comment:
My understanding was that this was a regression when switching to use `key`, although I'm not 100% sure why/how. I've refactored the commits so that this line is added in the commit that switches to using `key` in the first place, so as not to introduce the problem.

Reply:
Cool, glad to start having `autoFocusMessage`, `initialTopic`, and `initialMessage` depend on `editMessage` in this commit. In the commit message, let's write down our reason for making those changes in this commit, as discussed above (https://github.com/zulip/zulip-mobile/pull/4773/commits/4691ff273c7806a56e8319c03dad66ce693d7a7d#r648445186), so that it's clear and inspectable in the future. 🙂

Also, as I look again, I'm glad to have the mention of the "Warning: Cannot update a component from inside…" fix in the commit message, since this sure seems to be the commit that changes that behavior for the better. But while it's nice (and potentially really helpful for responding to bug reports) to know that the specific behavior is improved, it's also true that we didn't end up with a really precise diagnosis, and I think we should mention that. As-is, I think it's possible for someone to follow the link after "See the CZO thread" in hopes of finding a diagnosis, and they'll be disappointed and possibly confused when they don't find one.

Reply:
Added a comment about this in the commit message. I don't think it's as mysterious as our initial conversation made it out to be — there was some logic in `UNSAFE_componentWillReceiveProps`, and those props are a way to replicate that in a non-imperative way, but it still seems worth mentioning. Also updated the bit about the error message disappearing.

Reply:
> I don't think it's as mysterious as our initial conversation made it out to be

Understood. 🙂 But we depend heavily on reading a clear commit history, and I think I've mentioned elsewhere that the `key` change is somewhat likely to introduce puzzling regressions, almost no matter how carefully we do it. So it'll be nice to have these details at hand if that happens, and not have to guess as much about why we made certain changes and whether the reasons were good based on what we knew at the time.

Also, I think explaining these prop changes does more than just answer why we made them: it clarifies a bit the limits of what we considered. Without these explanations, I think someone in the future is more likely to assume that we deduced that these changes were _all_ we had to do toward replicating the logic in `UNSAFE_componentWillReceiveProps` in a non-imperative way. We didn't deduce that; we started with some prominent behaviors that we wanted to keep, without considering some small-looking things at all, like a `textInput.setNativeProps` call (whatever that's about) and everything that `TextInputReset` does. I think that's probably an OK tradeoff, so that we don't spend too long on this. But it is a tradeoff, and I think it'll be more obvious that we made the tradeoff after listing these things that we did do something about.